packages/fluentui/react-northstar/src/components/Carousel/CarouselNavigation.tsx

```diff
@@ -50,76 +62,123 @@ export interface CarouselNavigationProps extends UIComponentProps, ChildrenCompo
   vertical?: boolean;
 }
 
-export const carouselNavigationClassName = 'ui-carousel__navigation';
-
-class CarouselNavigation extends UIComponent<WithAsProp<CarouselNavigationProps>> {
-  static displayName = 'CarouselNavigation';
+export type CarouselNavigationStylesProps = Required<
+  Pick<CarouselNavigationProps, 'activeIndex' | 'iconOnly' | 'primary' | 'vertical' | 'thumbnails'>
+>;
 
-  static deprecated_className = carouselNavigationClassName;
+export const carouselNavigationClassName = 'ui-carousel__navigation';
 
-  static create: ShorthandFactory<CarouselNavigationProps>;
+export const CarouselNavigation: React.FC<WithAsProp<CarouselNavigationProps>> &
+  FluentComponentStaticProps<CarouselNavigationProps> = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(CarouselNavigation.displayName, context.telemetry);
+  setStart();
+  const {
+    accessibility,
+    variables,
+    children,
+    className,
+    design,
+    activeIndex,
+    iconOnly,
+    items,
+    primary,
+    secondary,
+    vertical,
+    thumbnails,
+    styles,
+  } = props;
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(CarouselNavigation.handledProps, props);
+
+  const getA11yProps = useAccessibility(accessibility, {
+    debugName: CarouselNavigation.displayName,
+    rtl: context.rtl,
```

Comment:
If we don't have any props, anyway define behavior for the component with `never` value and use it in the typings and both in the accessibility